wallet/src/de/schildbach/wallet/service/CoinJoinService.kt

```diff
@@ -46,66 +49,65 @@ import org.bitcoinj.coinjoin.utils.CoinJoinManager
 import org.bitcoinj.core.AbstractBlockChain
 import org.bitcoinj.core.Coin
 import org.bitcoinj.core.Context
+import org.bitcoinj.core.ECKey
 import org.bitcoinj.core.MasternodeAddress
+import org.bitcoinj.utils.ContextPropagatingThreadFactory
 import org.bitcoinj.utils.Threading
 import org.bitcoinj.wallet.Wallet
 import org.bitcoinj.wallet.WalletEx
+import org.bouncycastle.crypto.params.KeyParameter
 import org.dash.wallet.common.WalletDataProvider
 import org.dash.wallet.common.data.NetworkStatus
 import org.dash.wallet.common.services.BlockchainStateProvider
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 import javax.inject.Singleton
 
 enum class CoinJoinMode {
-    BASIC,
+    NONE,
     INTERMEDIATE,
-    ADVANCED,
+    ADVANCED
 }
 
+/**
+ * CoinJoin Service
+ *
+ * Monitor the status of the CoinJoin Mixing Service
+ */
 interface CoinJoinService {
-    var mode: CoinJoinMode
     val mixingStatus: MixingStatus
-
-    fun needsToMix(amount: Coin): Boolean
-    suspend fun configureMixing(
-        amount: Coin,
-        requestKeyParameter: RequestKeyParameter,
-        requestDecryptedKey: RequestDecryptedKey,
-        restoreFromConfig: Boolean
-    )
-
-    suspend fun prepareAndStartMixing()
-    suspend fun waitForMixing()
-    suspend fun waitForMixingWithException()
+    val mixingProgress: Flow<Double>
```

Comment:
The interface has been simplified to the point where this is used only for viewing progress.